coffee_lib/src/repository.rs

```diff
@@ -33,5 +33,14 @@ pub trait Repository: Any {
     /// return the url of the repository.
     fn url(&self) -> URL;
 
+    /// return the git head of the repository.
+    fn git_head(&self) -> Option<String>;
+
+    /// return the date of the last commit of the repository.
+    fn date_of_the_last_commit(&self) -> Option<String>;
+
+    /// return the git information of the repository.
+    async fn git_info(&self) -> Result<(String, String), CoffeeError>;
+
```

Comment:
If you noted inside the repository we do not have any reference to git, so the coffee_lib specifies only the central concept of the repository, but the repository is not coupled with git at all.

The git implementation is the coffee_github one, and where these methods should live too